src/components/Form/index.js

```diff
@@ -154,24 +161,50 @@ export default class Form extends Component {
             ),
             touched: true
          }
+
+         // Keep track of files on state.data
+         if (formFileComponentNames.includes(node.type.name)) {
+            namesForFiles.push(node.props.name)
+         }
+      })
+
+      const formData = new FormData()
+      // Apply "files" to formData
+      namesForFiles.forEach((name) => {
+         const file = newState.data[name].value
+         if (file) (formData.append(name, new Blob([file], { type: file.type }), file.name))
+      })
+      // Apply rest of data to formData
+      Object.keys(newState.data).forEach((key) => {
+         // Only apply if other data
+         if (!namesForFiles.includes(key)) {
+            // Append key to formData
+            formData.append(key, newState.data[key].value)
+         }
       })
 
       this.setState(newState, () => {
+         const data = {}
+         const errors = {}
+
          // Find data and errors
          Object.keys(this.state.data).forEach((key) => {
-            // Add the key/value to data
-            data[key] = this.state.data[key].value
-            // Add the error if applicable
-            if (this.state.data[key].error) {
-               errors[key] = this.state.data[key].error
+            // Don't add data keys that relate to files
+            if (!namesForFiles.includes(key)) {
+               // Add the key/value to data
+               data[key] = this.state.data[key].value
+               // Add the error if applicable
+               if (this.state.data[key].error) {
+                  errors[key] = this.state.data[key].error
+               }
             }
          })
 
          // Pass data or errors in to appropriate event handler prop
          if (Object.keys(errors).length > 0) {
             if (this.props.onError) this.props.onError(errors)
          } else if (this.props.onSubmit) {
-            const promise = this.props.onSubmit(data)
+            const promise = this.props.onSubmit(data, formData)
```

Comment:
Just curious, what's the reasoning in passing both `data` and `formData`, rather than just `formData`, which would have both file data and field data? Also in the case of a form with a file upload component, aren't you passing virtually duplicate data with the two params?